schemas/2018-05-01/deploymentTemplate.json

```diff
@@ -0,0 +1,477 @@
+        },
+        "allowedValues": {
+          "type": "array",
+          "description": "Value can only be one of these values"
+        },
+        "metadata": {
+          "type": "object",
+          "description": "Metadata for the parameter, can be any valid JSON object"
+        },
+        "minValue": {
+          "type": "integer",
+          "description": "Minimum value for the int type parameter"
+        },
+        "maxValue": {
+          "type": "integer",
+          "description": "Maximum value for the int type parameter"
+        },
+        "minLength": {
+          "type": "integer",
+          "description": "Minimum length for the string or array type parameter"
+        },
+        "maxLength": {
+          "type": "integer",
+          "description": "Maximum length for the string or array type parameter"
+        }
+      },
+      "required": [ "type" ],
+      "description": "Input parameter definitions"
+    },
+    "output": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "$ref": "#/definitions/parameterTypes",
+          "description": "Type of output value"
+        },
+        "value": {
+          "$ref": "#/definitions/parameterValueTypes",
+          "description": "Value assigned for output"
+        }
+      },
+      "required": [ "type", "value" ],
+      "description": "Set of output parameters"
+    },
+    "parameterTypes": {
+      "enum": [ "string", "securestring", "int", "bool", "object", "secureObject", "array" ]
+    },
+    "parameterValueTypes": {
+      "type": [ "string", "boolean", "integer", "number", "object", "array", "null" ]
+    },
+    "keyVaultReference": {
+      "type": "object",
+      "properties": {
+        "keyVault": {
+          "type": "object",
+          "properties": {
+            "id": {
+              "type": "string",
+              "minLength": 1
+            }
+          },
+          "required": [ "id" ],
+          "additionalProperties": false
+        },
+        "secretName": {
+          "type": "string",
+          "minLength": 1
+        },
+        "secretVersion": {
+          "type": "string",
+          "minLength": 1
+        }
+      },
+      "required": [
+        "keyVault",
+        "secretName"
+      ],
+      "additionalProperties": false
+    },
+    "expression": {
+      "type": "string",
+      "pattern": "^\\[(json|if|and|or|not|concat|parameters|variables|deployment|reference|resourceId|resourceGroup|subscription|list.*|base64|providers|copyIndex|padLeft|replace|toLower|toUpper|startsWith|endsWith|length|split|skip|take|contains|intersection|union|first|last|indexOf|lastIndexOf|add|sub|mul|div|mod|min|max|range|string|int|float|bool|trim|uri|uniqueString|substring|base64ToString|base64ToJson|uriComponentToString|uriComponent|dataUriToString|dataUri|array|createArray|coalesce|empty|less|lessOrEquals|greater|greaterOrEquals|equals|[\\w\\$]+\\.[\\w\\$]+)\\(.*\\).*\\]$",
```

Comment:
I would be in favor of removing these as well so we don't have a maint problem and vs/vscode do not use this... (even though others may)

Reply:
I have the impression that vs/code use it. if you write an expression which doesn't follow this. You get a validation error.

Reply:
no, vs/code maintain their own list: https://github.com/bmoore-msft/vscode-azurearmtools/blob/master/assets/ExpressionMetadata.json

Reply:
They aren't doing anything for UDFs currently...

Reply:
In that case should we keep the expression definition here, until they support UDF?

Reply:
Actually UDF is a good point... since we don't support UDF in the schema, we'll be flagging those as invalid.  IOW, since the user's UDF won't match the regex, this regex will flag those as invalid.  So we need to remove these and just use [*] for the expression regex

Reply:
no the current expression supports UDF. If customer specifies util.Max() we accept it as a valid expression.